clang/bindings/python/clang/cindex.py

```diff
@@ -64,48 +64,80 @@
 
 from ctypes import *
 
-import collections.abc
 import os
+import sys
 from enum import Enum
 
+from typing import (
+    Any,
+    Callable,
+    Generic,
+    Optional,
+    Type as TType,
+    TypeVar,
+    TYPE_CHECKING,
+    Union as TUnion,
+)
+
+if TYPE_CHECKING:
+    from ctypes import _Pointer
+    from typing_extensions import Protocol, TypeAlias
```

Comment:
IIRC `typing_extensions` are not in the Python standard library, and we didn't use them before. Which make them a novel dependency. At the very least this should be mentioned in the release notes. Otherwise I'm not sure how `cindex.py` appears on users' systems, or whether we maintain an official package with a list of dependencies.

Reply:
Yes that's right, that's why I put this inside the `if TYPE_CHECKING` guard: this line will only be executed on a type checker run, so it does not affect normal usage at all. The CI would fail otherwise, and in fact, did, before I moved this inside that block.
Does this still need a release note in that case?

Reply:
Sure, this is not going to affect normal usage. But I still wonder how it's handled in the ecosystem when `TYPE_CHECKING` mode brings additional dependencies. If `typing_extensions` are generally assumed to be available in this mode, then I'm fine with it.

Reply:
There is no strict requirement for `typing_extensions` to be available, though all type checkers I've seen are either written in Python and depend on `typing_extensions` themselves  (the default checker `mypy`, Google's `pytype`, Facebook's `pyre`), or don't use Python at all (Microsoft's `pyright`) in which case the installed modules don't matter
Aside from that, as `TYPE_CHECKING` mode doesn't actually include any execution, the module not being installed would never lead to an error. At most, the type checker would complain that it can't resolve the import.